src/library/baseplaylistfeature.cpp

```diff
@@ -633,17 +633,17 @@ QModelIndex BasePlaylistFeature::constructChildModel(int selected_id) {
         }
 
         // Create the TreeItem whose parent is the invisible root item
-        TreeItem* item = new TreeItem(playlistLabel, playlistId);
-        item->setBold(m_playlistsSelectedTrackIsIn.contains(playlistId));
+        std::unique_ptr<TreeItem> pItem = std::make_unique<TreeItem>(playlistLabel, playlistId);
+        pItem->setBold(m_playlistsSelectedTrackIsIn.contains(playlistId));
 
-        decorateChild(item, playlistId);
-        data_list.append(item);
+        decorateChild(pItem.get(), playlistId);
```

Comment:
`decorateChild` seems to be trivial to refactor to use a `gsl::not_null<TreeItem>`. That would help clarify that its non-owning (at least I think we established that `gsl::not_null` pointers are non-owning, didn't we?)

Reply:
gsl is unfortunately not available in the 2.3 branch.

Reply:
Mhmm, then it would be nice if you could document the non-owning nature of pointer in `BasePlaylistFeature`

Reply:
Done.